### PR TITLE
feat(2767): Add stage api

### PR DESCRIPTION
**Pull request by tkyi**
## Context

Split up the API PR into API and workflow.

## Objective

This PR adds stage API endpoints.

## References

Related to https://github.com/screwdriver-cd/screwdriver/issues/2767

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/pipelines/README.md [hunk 1]
```diff
@@ -121,9 +121,7 @@ Only PR events of specified PR number will be searched when `prNum` is set
 #### Get all stages for a single pipeline
 
 `GET /pipelines/{id}/stages`
-Will get latest commit event's stages if no event ID or group event ID is provided
-
-`GET /pipelines/{id}/stages?groupEventId={groupEventId}`
+Will get latest commit event's stages if no event ID is provided
```
Comment (sagar1312) [inline plugins/pipelines/README.md:124]: With the latest schema changes, `stages` are no longer associated to `event`. I believe `GET /pipelines/{id}/stages?eventId={eventId}` is no longer valid endpoint.
```suggestion
```

**Review by sagar1312 — CHANGES_REQUESTED** [review]

**Review by sagar1312 — CHANGES_REQUESTED** [review]
Also remove `GET /stages/{id}/stageBuilds?eventId={eventId}` from `plugins/stages/README.md`

Not entirely related to this PR. It will nice to have some test to validate all the referenced factories exists.
Ex: New factory `stageBuildFactory` is being referenced which is not yet available in the latest `models` dependency.

File: test/plugins/events.test.js [hunk 2]
```diff
@@ -223,6 +241,49 @@ describe('event plugin test', () => {
...
+        let stageBuilds;
+
+        beforeEach(() => {
+            options = {
+                method: 'GET',
+                url: `/events/${id}/stageBuilds`
+            };
+
+            event = getEventMock(testEvent);
+            stageBuilds = getStageBuildMocks(testStageBuilds);
```
[the remark this answers is not in the document]
Comment (tkyi) [reply test/plugins/events.test.js:257]: removed from models

File: plugins/helper.js [hunk 3]
```diff
@@ -102,10 +103,21 @@ async function getScmUri({ pipeline, pipelineFactory }) {
     return scmUri;
 }
 
+/**
+ * Returns full stage name with correct formatting and setup or teardown suffix (e.g. stage@deploy:setup)
+ * @param  {String} stageName               Stage name
+ * @param  {String} type                    Type of stage job, either 'setup' or 'teardown'
+ * @return {String}                         Full stage name
+ */
+function getFullStageJobName({ stageName, jobName }) {
```
[the remark this answers is not in the document]
Comment (tkyi) [reply plugins/helper.js:112]: removed

**Review by sagar1312 — COMMENTED** [review]
Looks good. I believe, this PR should be merged after https://github.com/screwdriver-cd/models/pull/570 goes in.

Note: It will be nice to explicitly upgrade to the `screwdriver-models` version required in the `package.json` once it is available.

**Review by sagar1312 — APPROVED** [review]

**Pull request merged**